pulumitest/newStack.go

```diff
@@ -171,6 +173,13 @@ func (pt *PulumiTest) NewStack(stackName string, opts ...optnewstack.NewStackOpt
 	if !stackOptions.SkipDestroy {
 		pt.t.Cleanup(func() {
 			pt.t.Helper()
+
+			if ptFailed(pt.t) {
+				pt.t.Log(fmt.Sprintf("refusing to destroy stack at %q to help debug the failing test",
+					stack.Workspace().WorkDir()))
+				return
+			}
```

Comment:
I was just writing this - most providers do not have cleanup scripts, so this would cause a lot of leaks.

Reply:
I'm reasonably certain this behavior shouldn't be opt-in but the default behavior, following ProgramTest precedent; the failing tests absolutely nasty to debug and keeping the information around on failing runs is a first good step to giving a tired developer a chance to make sense of the failing tests.

Reply:
>opt-in but the default behavior

Happy with this behaviour for local dev but I don't see what it brings in CI - the runners will likely clean it up. 

Alternatively it could be a provider flag, which we can roll out across all bridged providers.

Reply:
>as tests aren't guaranteed to complete (CI terminated) 

The reality is that most cloud services don't have a cleanup scripts and these are somewhat involved to get working properly as evidenced by the AWS one

Reply:
I'm not sure I fully understand the reasoning about racking up costs @VenelinMartinov. Don't our tests run in GitHub actions? So any file created during the test run should be thrown away after it, right?

Reply:
They provision real cloud resources, which cost us money in the respective clouds.

Reply:
We handle this in AWS with a nightly cleanup script which deletes resources but we don't have anythng like that for less popular service providers.

Reply:
I think this is reasonable default behavior for local development but agree we should try as hard as possible to cleanup in CI. The service already struggles with the amount of garbage data we've left lingering, and I don't want to exacerbate that.

Suggest something like this for the best of both worlds:

```diff
-if ptFailed(pt.t) {
+if ptFailed(pt.t) && !ciutil.IsCI() {
```

Reply:
Sounds good, I'll take this!

Reply:
Addressed this thanks y'all.